core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -4768,6 +4768,112 @@ public String getDefaultScheme() {
+                      + "\"name\":[{\"service\":\"SimpleService1\"}],"
+                      + "\"waitForReady\":true}]}");
+      channelBuilder.defaultServiceConfig(defaultServiceConfig);
+      ManagedChannelServiceConfig managedChannelServiceConfig =
+          createManagedChannelServiceConfig(defaultServiceConfig, null);
+      nameResolverFactory.nextConfigOrError.set(
+          ConfigOrError.fromConfig(managedChannelServiceConfig));
+
+      createChannel(true);
+      int prevSize = getStats(channel).channelTrace.events.size();
+
+      nameResolverFactory.resolvers.get(0).listener.onError(resolutionError);
+
+      assertEquals(channel.getConfigSelector().getClass().getName(),
+          managedChannelServiceConfig.getDefaultConfigSelector().getClass().getName());
+      assertEquals(channel.getLastServiceConfig().toString(),
+          managedChannelServiceConfig.toString());
+
+      assertThat(getStats(channel).channelTrace.events).contains(new ChannelTrace.Event.Builder()
+              .setDescription("Initial Name Resolution error, using default service config")
+              .setSeverity(ChannelTrace.Event.Severity.CT_ERROR)
+              .setTimestampNanos(timer.getTicker().read())
+              .build());
+      assertThat(getStats(channel).channelTrace.events.get(prevSize - 1))
+              .isEqualTo(new ChannelTrace.Event.Builder()
+              .setDescription("Failed to resolve name: " + resolutionError)
+              .setSeverity(ChannelTrace.Event.Severity.CT_WARNING)
+              .setTimestampNanos(timer.getTicker().read())
+              .build());
+
+      // no new trace events due to lastResolutionState already set to ERROR
+      assertThat(getStats(channel).channelTrace.events).hasSize(prevSize);
+    } finally {
+      LoadBalancerRegistry.getDefaultRegistry().deregister(mockLoadBalancerProvider);
+    }
+  }
+
+  @Test
+  public void testTraceEventsOnResultsCallFollowedByOnErrorCallWithAppliedServiceConfig()
+          throws Exception {
+    LoadBalancerRegistry.getDefaultRegistry().register(mockLoadBalancerProvider);
+    timer.forwardNanos(1234);
+    channelBuilder.maxTraceEvents(10);
+    try {
+      List<EquivalentAddressGroup> servers = new ArrayList<>();
+      servers.add(new EquivalentAddressGroup(socketAddress));
+      FakeNameResolverFactory nameResolverFactory =
+          new FakeNameResolverFactory.Builder(expectedUri).setServers(servers).build();
+      channelBuilder.nameResolverFactory(nameResolverFactory);
+      Map<String, Object> defaultServiceConfig =
+          parseConfig("{\"methodConfig\":[{"
+              + "\"name\":[{\"service\":\"SimpleService1\"}],"
+              + "\"waitForReady\":true}]}");
+      channelBuilder.defaultServiceConfig(defaultServiceConfig);
+      ManagedChannelServiceConfig managedChannelServiceConfig =
+          createManagedChannelServiceConfig(defaultServiceConfig, null);
+      nameResolverFactory.nextConfigOrError.set(
```

Comment:
Same question here, doesn't seem to  get used .

Reply:
This will give the info like we are getting the defaultServiceConfig details which are internally setting to through ManagedChannelServiceConfig and setting this into nameResolverFactory which we are using below.

defaultServiceConfig -> ManagedChannelServiceConfig -> nameResolverFactory.nextConfigOrError.

do you thinks its not required? because I have seen some existing references similar to this in existing UT's in the same class.

Reply:
Fixed and Removed this setup/config step as it's not harming test flow.